homeassistant/components/wemo/__init__.py

```diff
@@ -90,23 +93,26 @@ async def async_setup(hass, config):
     return True
 
 
-async def async_setup_entry(hass, entry):
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up a wemo config entry."""
     config = hass.data[DOMAIN].pop("config")
 
     # Keep track of WeMo device subscriptions for push updates
     registry = hass.data[DOMAIN]["registry"] = pywemo.SubscriptionRegistry()
     await hass.async_add_executor_job(registry.start)
 
+    wemo_dispatcher = WemoDispatcher(entry)
+    wemo_discovery = WemoDiscovery(hass, wemo_dispatcher)
+
+    @callback
     def stop_wemo(event):
         """Shutdown Wemo subscriptions and subscription thread on exit."""
         _LOGGER.debug("Shutting down WeMo event subscriptions")
         registry.stop()
```

Comment:
Is this async safe?

Reply:
They are intended to be async safe. Did I miss something?

Reply:
Oh, sorry, I got confused with the diff. You're probably asking about `registry.stop()`. I don't think that's async safe. Fixed.